puppetboard/app.py

```diff
@@ -399,205 +408,122 @@ def node(env, node_name):
 
     node = get_or_abort(puppetdb.node, node_name)
     facts = node.facts()
-    reports = get_or_abort(puppetdb.reports,
-                           query=query,
-                           limit=app.config['REPORTS_COUNT'],
-                           order_by=DEFAULT_ORDER_BY)
-    reports, reports_events = tee(reports)
-    report_event_counts = {}
-
-    for report in reports_events:
-        report_event_counts[report.hash_] = {}
-
-        for event in report.events():
-            if event.status == 'success':
-                try:
-                    report_event_counts[report.hash_]['successes'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['successes'] = 1
-            elif event.status == 'failure':
-                try:
-                    report_event_counts[report.hash_]['failures'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['failures'] = 1
-            elif event.status == 'noop':
-                try:
-                    report_event_counts[report.hash_]['noops'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['noops'] = 1
-            elif event.status == 'skipped':
-                try:
-                    report_event_counts[report.hash_]['skips'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['skips'] = 1
     return render_template(
         'node.html',
         node=node,
         facts=yield_or_stop(facts),
-        reports=yield_or_stop(reports),
-        reports_count=app.config['REPORTS_COUNT'],
-        report_event_counts=report_event_counts,
         envs=envs,
-        current_env=env)
-
-
-@app.route('/reports/',
-           defaults={'env': app.config['DEFAULT_ENVIRONMENT'], 'page': 1})
-@app.route('/<env>/reports/', defaults={'page': 1})
-@app.route('/<env>/reports/page/<int:page>')
-def reports(env, page):
-    """Displays a list of reports and status from all nodes, retreived using the
-    reports endpoint, sorted by start_time.
+        current_env=env,
+        table_count=app.config['LITTLE_TABLE_COUNT'],
+        table_count_selector=app.config['TABLE_COUNT_SELECTOR'],
```

Comment:
The templates have direct access to the config settings because it's rendered in the Flask application's context. `app.config['LITTLE_TABLE_COUNT']` and `app.config['TABLE_COUNT_SELECTOR']` are available in the templates as `config.LITTLE_TABLE_COUNT` and `config.TABLE_COUNT_SELECTOR` respectively.